openerp/openupgrade/openupgrade_70.py

```diff
@@ -42,14 +41,18 @@ def set_partner_id_from_partner_address_id(
     """
     model = pool.get(model_name)
     table = table or model._table
-    # Cannot use cursor's string substitution for table names
-    cr.execute("""
-        SELECT target.id, address.openupgrade_7_migrated_to_partner_id
-        FROM %s as target,
-             res_partner_address as address
-        WHERE address.id = target.%s""" % (table, address_field))
-    for row in cr.fetchall():
-        model.write(cr, SUPERUSER_ID, row[0], {partner_field: row[1]})
+    openupgrade.logged_query(
+        cr,
+        "UPDATE %s "
+        "SET %s=src.partner_id "
+        "FROM ("
+        "    SELECT target.id as id, "
+        "           address.openupgrade_7_migrated_to_partner_id as partner_id"
+        "    FROM %s as target, res_partner_address as address "
+        "    WHERE address.id = target.%s"
+        ") src "
+        "WHERE id=src.id" %
+        (table, partner_field, table, address_field))
```

Comment:
You can eliminate the subquery using

UPDATE %{table]s
SET %{partner_field}s=address.openupgrade_7_migrated_to_partner_id
FROM res_partner_address address
WHERE %{table}s.%{address_field}=address.id